File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        
    Set<Integer> numSet = new HashSet<>();
    for (int num : nums) numSet.add(num);
    
        int max = 0;
        for(int i = 0; i< nums.length;i++) {
            if (!numSet.contains((nums[i] - 1))) {
                int temp = nums[i];
                int t = 0;
                while (numSet.contains(temp)) {
                    temp++;
                    t++;
                }
                
                    max = Math.max(max, t);
            }
        }

        return max;
    }
}

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-1.kt
class Solution {
    fun lengthOfLIS(nums: IntArray): Int {
        val dp = IntArray ( nums.size ) {1}
        for (i in 0..nums.size - 1) {
            for (j in 0..i-1) {
                if (nums[j] < nums[i]) dp[i] = maxOf(dp[i], dp[j]+1)
            }
        }

        return dp.max()
    }
}
